### ISSUE TITLE
Broken due to the Devotion Update

### ISSUE THREAD

**Issue by Moffein** [ThinkInvis/RoR2-Yeet#18]
### Describe the Bug
When tossing an item, it stays as the in-flight sphere thing and rolls around on the ground instead of turning into an item on collision.
![20240601141824_1](https://github.com/ThinkInvis/RoR2-Yeet/assets/22742925/332d3bec-ff76-446a-ae26-c9b7037a3c25)

### To Reproduce
1. Toss an item.
2. The item hits the ground and rolls around as a sphere.

### Expected Behavior
Item is dropped normally.

### Console Log
Error is at the end of the log.
[LogOutput.log](https://github.com/user-attachments/files/15523408/LogOutput.log)

### Additional Context
Probably should update the dlls used to compile the mod as well, since I recall some stuff changing related to PickupDropletControllers.
Tried running the mod with GoxOfBears (fixes PickupDropletController stuff), but the issue still persisted.

The specific nullref is 
```[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ThinkInvisible.Yeet.YeetPlugin.PickupDropletController_OnCollisionEnter (On.RoR2.PickupDropletController+orig_OnCollisionEnter orig, RoR2.PickupDropletController self, UnityEngine.Collision collision) (at <937d2afa57214dbe9e16dd7992b14e56>:IL_00AC)
DMD<>?-1737733888.Hook<RoR2.PickupDropletController::OnCollisionEnter>?1916794240 (RoR2.PickupDropletController , UnityEngine.Collision ) (at <96457fedccf2451597afd0e44a6a5847>:IL_0014)```

**Comment by FlyAndNotDown** [comment]
Seems I get the same problem, anyone can fix this issue ?

**Comment by ThinkInvis** [comment]
Fixed by 41f3b93089d6a70ffb2cd2994a96e277e5b6c98f.

**Comment by DistantJragon** [comment]
I'm getting the same issue as of the release of the update coinciding with the Alloyed Collective DLC.

[LogOutputYeet.log](https://github.com/user-attachments/files/23641561/LogOutputYeet.log)

```
[Error  : Unity Log] MissingMethodException: Method not found: void RoR2.GenericPickupController.set_NetworkpickupIndex(RoR2.PickupIndex)
Stack trace:
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Hook<RoR2.PickupDropletController::OnCollisionEnter>?1791140908(RoR2.PickupDropletController,UnityEngine.Collision) (at IL_001F)
```
